node/node.go

```diff
@@ -1203,8 +1203,7 @@ func (node *AlgorandFullNode) SetCatchpointCatchupMode(catchpointCatchupMode boo
 			node.txPoolSyncerService.Stop()
 			node.blockService.Stop()
 			node.ledgerService.Stop()
-			// As for now, the state proof services handles catchpoint mode
-			// so we don't need to stop it.
+			node.stateProofWorker.Shutdown()
```

Comment:
This call closes the db created in `MakeFull`:
`	stateProofAccess, err := db.MakeAccessor(stateProofPathname, false, false)`

But when you call `Start` again, you use a closed accessor:
`	err := makeStateProofDB(spw.db)`

What am I missing here?
If I am not missing anything, you are missing a test :-)

Reply:
appears we need both Stop and Shutdown, and close the assessor only in the latter, or move `stateProofAccess, err := db.MakeAccessor(stateProofPathname, false, false)` to `Start`

Reply:
I prefer option 2: just have Start/Stop, and move accessor creation to `Start`.

Reply:
agree with Shant.